README.md

```diff
@@ -1,28 +1,78 @@
+        uses: tibdex/github-app-token@v1
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.PRIVATE_KEY }}
+      - name: Checkout Code
+      - uses: actions/checkout@v3
       - name: Detect unused secrets
-        uses: nearform/github-action-detect-unused-secrets@v1
+        uses: nearform/github-action-detect-unused-secrets@main
         with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
+          github-token: ${{ steps.generate_token.outputs.token }}
+
 ```
 
 ## Usage
 
 Configure this action in your workflows providing the inputs described below.
 
+### Install a GitHub App
+
+**Required** [Create a GitHub App](https://docs.github.com/en/developers/apps/building-github-apps/creating-a-github-app) within your organisation with permission to `secrets: read-only` and [install](https://docs.github.com/en/developers/apps/managing-github-apps/installing-github-apps#installing-your-private-github-app-on-your-repository) it on the repo where you want to use this action. Save the `APP_ID` and generated `PRIVATE_KEY` to the secrets under your organisation or repo.
+
+This action uses the [listRepoSecrets endpoint](https://docs.github.com/en/rest/reference/actions#list-repository-secrets), your GitHub App must have the `secrets` repository permission to use this endpoint.
+
+> Note that you can skip this step if your repo is `not` under an organisation, you can [create a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) `with repo scope` and reference it as github-token
+
+### uses: tibdex/github-app-token@v1
+
+**Required**
+This step is used when `secrets.GITHUB_TOKEN`'s limitations are too restrictive and a personal access token is not suitable.
+
+The default `secrets.GITHUB_TOKEN` has limited permissions [click here for more information](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token), this step is generating a new token. `APP_ID` and `PRIVATE_KEY` are required inputs, which have been generated by installing a GitHub App on the previous step
+
+```yml
+- name: Generate token
+  id: generate_token
+  uses: tibdex/github-app-token@v1
+  with:
+    app_id: ${{ secrets.APP_ID }}
+    private_key: ${{ secrets.PRIVATE_KEY }}
+```
+
+> Note that you can skip this step if your repo is `not` under an organisation, you can [create a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) `with repo scope` and reference it as github-token
+
+### uses: actions/checkout@v3
+
+**Required** This action checks-out your repository under `$GITHUB_WORKSPACE`. For information check [actions/checkout](https://github.com/actions/checkout). This step is required because under the hood `detect-unused-secrets` is executing a script to grab the secret names used in `.github/workflows`.
```

Comment:
same comment as before, no need to explain implementation details. we just need to explain that we need access to the content of the workflow files